### ISSUE TITLE
RootMutationResolver doesn't honor mapping in codegen.yml file

### ISSUE THREAD

**Issue by SirensOfTitan** [dotansimha/graphql-code-generator#886]
**Describe the bug**
So I have a root mutation object on the server, and generate a resolver for all mutations using `graphql-codegen-typescript-resolvers`.  I using mapping to map for user from a `UserEntity` to a gql-codgen generated `User`.

There was a correctly generated resolver signature for the VerifyUserPayload (the result of the `verifyUser` mutation):

```typescript
export namespace VerifyUserPayloadResolvers {
  export interface Resolvers<
    Context = AppContext,
    TypeParent = VerifyUserPayload
  > {
    user?: UserResolver<UserEntity | null, TypeParent, Context>;
  }
}
```

... but the RootMutationResolvers doesn't actually use the UserEntity, but the User:
```typescript
export namespace RootMutationResolvers {
  export interface Resolvers<Context = AppContext, TypeParent = {}> {
    verifyUser?: VerifyUserResolver<
      VerifyUserPayload | null,
      TypeParent,
      Context
    >;
  }
}

// ... definition of `VerifyUserPayload`, a root type non-namespaced:
export interface VerifyUserPayload {
  user?: User | null;
}
```

**Environment:**
 - OS: macOS
 - Codegen: 0.14.1 for all versions
 - Node: 8.11.2

**Additional context**

**Comment by dotansimha** [comment]
@kamilkisiela

**Comment by SirensOfTitan** [comment]
@kamilkisiela: our schema uses a `Payload` type for each mutation result to ensure we can deprecate or add return values to a mutation result later on, so using your added test case as a basis our schema would look more like:
```graphql
        type Query {
          post: Post
        }
        type RootMutation {
          upvotePost(id: String!): UpvotePostPayload
        }

       type UpvotePostPayload {
          post: Post
        }

        type Post {
          id: String
        }
        
        schema {
          query: Query
          mutation: RootMutation
        }
   
```

.... in that case, I believe, the generated type for the `RootMutationResolvers` would have an incorrect signature for `upvotePost` as  UpvotePostPayload would not have the correct signature with `PostEntity` replacing it.

**Comment by kamilkisiela** [comment]
@SirensOfTitan https://github.com/dotansimha/graphql-code-generator/commit/b1bf6983522211a0432392ff3d8cea6b014e98dc

**Comment by dotansimha** [comment]
Fixed in 0.14.2 🎉
